### PR TITLE
Add ExpandPath w/ tests. Returns abs path for supplied path.

### PR DESCRIPTION
**Pull request by Veebers**
Provide a method that returns an absolute path from the given path.

This will be used (at least initially) to resolve https://bugs.launchpad.net/juju/+bug/1745031 by removing the requirement that the provided path be an absolute path.

### CODE REVIEW

**Comment by Veebers** [comment]
[GCE Cred bug: 1745031](https://trello.com/c/f1Gf4xgg/178-gce-cred-bug-1745031)

File: file_test.go [hunk 1]
```diff
@@ -73,6 +73,66 @@ func (*fileSuite) TestNormalizePath(c *gc.C) {
...
+		path: "~foobar/path",
+		err:  ".*" + utils.NoSuchUserErrRegexp,
+	}} {
+		c.Logf("test %d: %s", i, test.path)
+		actual, err := utils.ExpandPath(test.path)
+		if test.err != "" {
+			c.Check(err, gc.ErrorMatches, test.err)
+		} else {
+			c.Check(err, gc.IsNil)
+			c.Check(actual, gc.Equals, test.expected)
```
Comment (howbazaar) [inline file_test.go]: Perhaps also add...
```go
c.Check(filepath.IsAbs(actual) , jc.IsTrue)
```

**Comment by Veebers** [comment]
$$merge$$

File: file.go [hunk 2]
```diff
@@ -51,6 +53,23 @@ func NormalizePath(dir string) (string, error) {
...
 }
 
+// ExpandPath normalises a path and expands it to an absolute form (if it's
+// not already an absolute path).
+func ExpandPath(path string) (string, error) {
+	normPath, err := NormalizePath(path)
+	if err != nil {
+		return "", errors.Annotate(err, "unable to normalise file path")
+	}
+	if !filepath.IsAbs(normPath) {
```
Comment (rogpeppe) [inline file.go]: Why not just use filepath.Abs(normPath) directly? It's more likely to work correctly on platforms other than Unix.

Also, the docs should probably mention Normalize by name - it's not necessarily clear that "normalising" a path implies expanding the `~` prefix.

**Pull request merged**